instrumentation/trace/context_test.go

```diff
@@ -45,3 +46,21 @@ func TestPropagateContextRetainsValue(t *testing.T) {
 	require.Equal(t, "foo", propgatedTracingContext.name)
 	require.NotEmpty(t, propgatedTracingContext.requestId)
 }
+
+func TestTranslateToRequestAndBack(t *testing.T) {
+	ctx := NewContext(context.Background(), "foo")
+	ep, _ := FromContext(ctx)
+
+	request, _ := http.NewRequest("Get", "localhost", nil)
+	ep.ToRequest(request)
+
+	require.Equal(t, "foo", request.Header.Get(RequestTraceName))
+
+	fctx := NewFromRequest(context.Background(), request)
+	ep2, ok := FromContext(fctx)
+	require.True(t, ok)
+	require.Equal(t, ep.name, ep2.name)
+	require.Equal(t, ep.requestId, ep2.requestId)
+	// had to compare this "flat" as the now function of time adds a debug string for monotonic that i can't seem to clear otherwise
+	require.EqualValues(t, ep.created.UnixNano(), ep2.created.UnixNano(), "%s %s", ep.created, ep2.created)
```

Comment:
Maybe simpler to use `require.True(t, ep.created.Equal(ep2), "...some error string...")`
This will also let you get rid of the comment.